From cbd68966aa427647f7e5ba31a859f5b4941c7016 Mon Sep 17 00:00:00 2001
From: irii <iriijo@googlemail.com>
Date: Sun, 11 Sep 2016 16:51:37 +0200
Subject: [PATCH 12/57] *This fixes the torch light for cameras with hal 1

*This fix works on the htc m7ul

*Fixed line length
*Fixed unused parameter names
---
 .../camera/libcameraservice/CameraFlashlight.cpp     | 20 +++++++++++++++++++-
 1 file changed, 19 insertions(+), 1 deletion(-)

diff --git a/services/camera/libcameraservice/CameraFlashlight.cpp b/services/camera/libcameraservice/CameraFlashlight.cpp
index 830867e3e..3979056c5 100644
--- a/services/camera/libcameraservice/CameraFlashlight.cpp
+++ b/services/camera/libcameraservice/CameraFlashlight.cpp
@@ -495,6 +495,23 @@ status_t CameraHardwareInterfaceFlashControl::initializePreviewWindow(
     return device->setPreviewWindow(mSurface);
 }
 
+static void notifyCallback(int32_t, int32_t, int32_t, void*) {
+    /* Empty */
+}
+
+static void dataCallback(int32_t, const sp<IMemory>&, camera_frame_metadata_t*, void*) {
+    /* Empty */
+}
+
+static void dataCallbackTimestamp(nsecs_t, int32_t, const sp<IMemory>&, void*) {
+    /* Empty */
+}
+
+static void dataCallbackTimestampBatch(int32_t, const std::vector<HandleTimestampMessage>&,
+        void*) {
+    /* Empty */
+}
+
 status_t CameraHardwareInterfaceFlashControl::connectCameraDevice(
         const String8& cameraId) {
     sp<CameraHardwareInterface> device =
@@ -508,7 +525,8 @@ status_t CameraHardwareInterfaceFlashControl::connectCameraDevice(
     }
 
     // need to set __get_memory in set_callbacks().
-    device->setCallbacks(NULL, NULL, NULL, NULL, NULL);
+    device->setCallbacks(notifyCallback, dataCallback, dataCallbackTimestamp,
+            dataCallbackTimestampBatch, this);
 
     mParameters = device->getParameters();
 
-- 
2.11.0

